server/src/main/java/org/elasticsearch/cluster/coordination/ElectionStrategy.java

```diff
@@ -34,6 +34,17 @@ protected boolean satisfiesAdditionalQuorumConstraints(
         }
     };
 
+    /**
+     * Contains a result for whether a node may win an election and the reason if not.
+     */
+    public record NodeEligibility(boolean mayWin, String reason) {}
+
+    public static final NodeEligibility NODE_MAY_WIN_ELECTION = new NodeEligibility(true, "");
+    public static final NodeEligibility NODE_MAY_NOT_WIN_ELECTION = new NodeEligibility(
+        false,
+        "node is ineligible for election, not a voting node in the voting configuration"
+    );
```

Comment:
The explanation is more specific than a constant name, I suggest to rename to `NODE_MAY_NOT_WIN_EXCLUDED_BY_VOTING_CONFIGURATION`